src/regions/regions-list.html

```diff
@@ -101,12 +101,23 @@
                     </div>
                     <div if.bind="!(shape.deleted && shape.is_new) &&
                                    !(roi.shapes.size > 1 && !roi.show)"
-                        title="ROI: ${roi_id} - Shape: ${shape_id}"
+                        title="${'ROI: ' + roi_id + ' - Shape: ' + shape_id +
+                                 (!shape.permissions ||
+                                    (shape.permissions &&
+                                     shape.permissions.canAnnotate &&
```

Comment:
to simplify code maybe introduce a "can_do(shape)" method taking a shape as an argument

Reply:
I know it doesn't look nice but in templates I usually refrain from using methods. For one I'd have to define them in the corresponding js to be able to use them and then I cannot reuse them outside but the actual reason I avoid them is that js object properties are smartly reevaluated on change and updated by aurelia by means of using an observer underneath (when it parses the template expressions). Aurelia cannot decorate function so change can be missed. In this particular example there wouldn't be a side-effect I suspect because the permission wouldn't change but if it did the corresponding UI change would not happen because the function cannot be watched for change, only properties can be "decorated" with an observer that notes the change. do you want me to make a local function for the check?

Reply:
a local function will work to avoid duplication..